### ISSUE TITLE
Can't filter containers in spinach.yaml for deployments

### ISSUE THREAD

**Issue by 1stewart** [derailed/popeye#200]
<img src="https://raw.githubusercontent.com/derailed/popeye/master/assets/popeye_bug.png" align="right" width="100" height="auto"/>

<br/>
<br/>
<br/>

**Describe the bug**
Can't filter containers in spinach.yaml for deployments. Filtering by deployment FQN or code works as expected.

**To Reproduce**

1. Have a report with a deployment warning for no resources/limits on a container

![image](https://user-images.githubusercontent.com/18353411/134690918-8e695ba1-e4ff-4268-bfdb-3f4bc7c56796.png)

2. Create a spinach.yaml file with the following content

```
popeye:
  excludes:
    apps/v1/deployments:
      - name: linkerd/linkerd-controller
        codes:
          - 106
```

3. Run with the spinach.yaml passed in `.\popeye.exe --save --out html -l warn -f .\spinach.yaml`

As expected, the warn isn't there

![image](https://user-images.githubusercontent.com/18353411/134692142-3f8aa830-d321-41d2-b97b-613f0a431368.png)

4. Create a spinach.yaml file with the following content (adding the container)

```
popeye:
  excludes:
    apps/v1/deployments:
      - name: linkerd/linkerd-controller
        containers:
          - public-api
        codes:
          - 106
```

5. Run with the spinach.yaml passed in `.\popeye.exe --save --out html -l warn -f .\spinach.yaml`

6. Container isn't picked up by configuration, fails back to showing non-matched deployment

![image](https://user-images.githubusercontent.com/18353411/134692687-03716c3a-259b-4183-8f46-60bf9353ee6b.png)

**Expected behavior**
(I think) it should allow filtering to containers, unless this is a limitation of the deployments API and the way they're referenced in its spec. 

```
kubectl get deployment -n linkerd linkerd-controller -o yaml
apiVersion: apps/v1
kind: Deployment
metadata:
...
  name: linkerd-controller
  namespace: linkerd
...
spec:
...
  template:
...
    spec:
      containers:
      - args:
        - public-api
...
...
        name: public-api

```
**Versions (please complete the following information):**
 - OS: Windows 10
 - Popeye: 0.9.7 (also on 0.9.0)
 - K8s 1.20.0

**Additional context**
Ideally we don't want to filter the whole deployment or popeye code, as any other containers would require limits/requests

**Comment by sgardnerdell** [comment]
+1

Would be really nice to exclude codes on specific containers.  Perhaps this could be achieved by changing the spinach configuration schema to allow:

```
  excludes:
    v1/pods:
    - name: argocd/argo-redis-ha-haproxy
      containers:
      - name: config-init
        codes:
        - 106
```

**Comment by derailed** [comment]
Fixed v0.25.6
